libflux/flux-core/src/semantic/bootstrap.rs

```diff
@@ -35,32 +36,89 @@ pub struct Error {
     pub msg: String,
 }
 
-/// DocPackage represents the documentation for a package and its sub packages
+/// Doc is an enum that can take the form of any form of flux documentation through polymorphism.
```

Comment:
"take the form of any form" reads awkwardly. I'd suggest something less formal-sounding, like "The various types of things that can be documented" or something like that. That way, I can look at the enum and go "Oh, hum, comments aren't documented, I guess" (the silliness of comments being documented notwithstanding).